<launch>
  <!-- size of map, change the size in x, y, z according to your application -->
  <arg name="map_size_x" value="45.0"/>
  <arg name="map_size_y" value="45.0"/>
  <arg name="map_size_z" value="23.0"/>

  <arg name="init_x" value="0"/>
  <arg name="init_y" value="0"/>
  <arg name="init_z" value="1.0"/> 

  <!-- topic of your odometry such as VIO or LIO **/airsim_node/drone_1/odom_local_enu in AirSim** -->
  <arg name="odom_topic" value="/airsim_node/drone_1/odom_local_enu" />

  <!-- main algorithm params -->
  <include file="$(find exploration_manager)/launch/algorithm.xml">

    <arg name="map_size_x_" value="$(arg map_size_x)"/>
    <arg name="map_size_y_" value="$(arg map_size_y)"/>
    <arg name="map_size_z_" value="$(arg map_size_z)"/>

    <arg name="box_min_x" value="-5.0"/>
    <arg name="box_min_y" value="-22"/>
    <arg name="box_min_z" value=" 0.0"/>
    <arg name="box_max_x" value="39.0"/>
    <arg name="box_max_y" value="22"/>
    <arg name="box_max_z" value="22.5"/>

    <arg name="odometry_topic" value="$(arg odom_topic)"/>
    <!-- sensor pose: transform of camera frame in the world frame -->
    <arg name="sensor_pose_topic" value="/airsim_node/drone_1/odom_local_enu"/>
    <!-- depth topic: depth image, 640x480 by default -->
    <!-- cloud topic: point cloud measurement  -->
    <!-- subscribe ONLY TO ONE of the two topics -->
    <arg name="depth_topic" value="/airsim_node/drone_1/front_center/DepthPerspective"/>
    <!-- <arg name="depth_topic" value="/depth"/> -->
    <!-- <arg name="cloud_topic" value="/airsim_node/drone_1/lidar/LidarCustom"/> -->
    <arg name="cloud_topic" value="/cloud"/>

    <!-- intrinsic params of the depth camera -->
    <arg name="cx" value="320.0"/>
    <arg name="cy" value="240.0"/>
    <arg name="fx" value="554.2562866210938"/>
    <arg name="fy" value="554.2562866210938"/>

    <!-- maximum velocity and acceleration the drone will reach -->
    <arg name="max_vel" value="0.8" />
    <arg name="max_acc" value="0.8" />

    <!-- perception distance -->
    <arg name="see_dist" value="10.5" />
    <arg name="map_dist" value="11.0" />
  </include>

  <!-- trajectory server -->
  <node pkg="plan_manage" name="traj_server" type="traj_server" output="screen">
    <remap from="/position_cmd" to="planning/pos_cmd"/>

    <remap from="/odom_world" to="$(arg odom_topic)"/>
    <param name="traj_server/time_forward" value="1.5" type="double"/>
    <param name="traj_server/pub_traj_id" value="4" type="int"/>
    <param name="traj_server/init_x" value="$(arg init_x)" type="double"/>
    <param name="traj_server/init_y" value="$(arg init_y)" type="double"/>
    <param name="traj_server/init_z" value="$(arg init_z)" type="double"/>

    <param name="perception_utils/top_angle" value="0.56125" type="double"/>
    <param name="perception_utils/left_angle" value="0.69222" type="double"/>
    <param name="perception_utils/right_angle" value="0.68901" type="double"/>
    <param name="perception_utils/max_dist" value="6.5" type="double"/>
    <param name="perception_utils/vis_dist" value="1.0" type="double"/>
    

  </node>

  <node pkg="waypoint_generator" name="waypoint_generator" type="waypoint_generator" output="screen">
    <remap from="~odom" to="$(arg odom_topic)"/>        
    <remap from="~goal" to="/move_base_simple/goal"/>
    <remap from="~traj_start_trigger" to="/traj_start_trigger" />
    <param name="waypoint_type" value="point"/>    
  </node>

</launch>
